--- a/drivers/net/wireless/ath/dfs_pattern_detector.c
+++ b/drivers/net/wireless/ath/dfs_pattern_detector.c
@@ -299,6 +299,12 @@ dpd_add_pulse(struct dfs_pattern_detecto
 				"count=%d, count_false=%d\n",
 				event->freq, pd->rs->type_id,
 				ps->pri, ps->count, ps->count_falses);
+
+                        printk(KERN_DEBUG "dfs radar found: freq=%d, type=%d, pri=%d, "
+                                "count=%d, count_false=%d\n",
+                                event->freq, pd->rs->type_id,
+                                ps->pri, ps->count, ps->count_falses);
+
 			pd->reset(pd, dpd->last_pulse_ts);
 			return true;
 		}
